src/shared/services/filterPanelService.js

```diff
@@ -0,0 +1,34 @@
+import angular from 'angular';
+import { mockDocuments } from '../../assets/documents';
+
+const filterPanelModule = angular.module('searchPage');
+filterPanelModule.service('filterPanelService', () => {
+    const filteredArticles = [];
+
+    function getFilteredArticles() {
+        const cache = {};
+        mockDocuments.forEach((elem) => {
+            const {
+                category,
+            } = elem;
+            if (cache[category]) {
+                cache[category].push(elem.value);
+            } else {
+                cache[category] = [elem.value];
+            }
+        });
+        Object.keys(cache).forEach((key) => {
+            filteredArticles.push({
+                filteredCategoryName: key,
+                filteredGroup: cache[key],
+            });
+        });
+    }
+    getFilteredArticles();
```

Comment:
I think moving function call to controller and returning categories would be better

Reply:
Function call was moved  to controller to return values of categories.